healthcheck.php

```diff
@@ -0,0 +1,95 @@
+<?php
+namespace Automattic\VIP;
+
+/**
+ * Utility Healthcheck class that runs further than cache-healthcheck,
+ * and allows to verify the app code is working as expected.
+ */
+class Healthcheck {
+	/**
+	 * By default, an app is considered healthy unless filter value returns a non-empty array
+	 *
+	 * @var string[] - Either empty array (no errors) or array of strings: [ 'an error has occurred', 'another error' ].
+	 */
+	protected array $errors = [];
+
+	public function __construct() {
+		$this->check();
+	}
```

Comment:
Having side effects like this, within the constructor, makes the code significantly harder to add tests for. Constructors are there to set up the class, not run a WP filter function.

Wherever the class is instantiated, should call `->check()`, directly itself.